internal/operands/common-instancetypes/kustomize.go

```diff
@@ -0,0 +1,182 @@
+package common_instancetypes
+
+import (
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// This is a simple wrapper around the Krusty kustomize entry point allowing us to mock things out in unit tests
+type KustomizeWrapper struct {
```

Comment:
I'd prefer if you make that an interface with just the `Run(fSys filesys.FileSystem, path string) (resmap.ResMap, error)` method. This way you can pass in the real `krusty.Kustomizer` or a `MockKustomizer` instance and we don't need the current proxy `Run` method.